packages/react-magma-dom/src/components/Popover/PopoverContent.tsx

```diff
@@ -0,0 +1,251 @@
+  /* align-items: center; */
+  max-height: ${props => (props.maxHeight ? props.maxHeight : '100%')};
+`;
+
+const StyledPopoverContent = styled.div`
+  z-index: 2;
+`;
+
+const PopoverChildrenContent = styled.div``;
+
+const ScrollableContent = styled.div`
+  /* width: 100%; */
+  display: flex;
+  flex-direction: column;
+  /* flex-wrap: wrap; */
+  overflow-y: auto;
+  padding: 16px;
+  height: inherit;
+`;
+
+function isExistedActiveElements(ref) {
+  return Array.from(
+    ref.current?.querySelectorAll(
+      'button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"]), video'
+    ) || []
+  ).filter((element: HTMLElement) => {
+    const style = window.getComputedStyle(element);
+    return (
+      element instanceof HTMLElement &&
+      style.display !== 'none' &&
+      style.visibility !== 'hidden' &&
+      !element.hasAttribute('disabled')
+    );
+  }).length > 0;
+}
+
+export const PopoverContent = React.forwardRef<
+  HTMLDivElement,
+  PopoverContentProps
+>((props, forwardedRef) => {
+  const { children, testId, ...other } = props;
+  const context = React.useContext(PopoverContext);
+  const theme = React.useContext(ThemeContext);
+  const ref = useForkedRef(forwardedRef, context.contentRef);
+
+  const focusTrapRef = useFocusLock(context.isOpen && isExistedActiveElements(context.contentRef));
+
+  const styledChildren = React.Children.toArray(children).map(item =>
+    React.cloneElement(item as React.ReactElement, {
+      theme,
+      isInverse: context.isInverse,
+    })
+  );
+  const header = styledChildren.find(
+    child =>
+      (child as React.ReactElement).type &&
+      (child as React.ReactElement).type === PopoverHeader
+  );
+  const footer = styledChildren.filter(
+    child =>
+      !(child as React.ReactElement).type ||
+      (child as React.ReactElement).type === PopoverFooter
+  );
+  const content = styledChildren.filter(
+    child =>
+      !(child as React.ReactElement).type ||
+      ((child as React.ReactElement).type !== PopoverHeader &&
+        (child as React.ReactElement).type !== PopoverFooter)
+  );
+
+  return (
+    <StyledPopoverContent
+      ref={context.setFloating}
+      style={{ ...context.floatingStyles }}
+    >
+      <StyledCard
+        {...other}
+        positioning={context.positioning}
+        hasDropShadow
+        isInverse={context.isInverse}
+        isOpen={context.isOpen}
+        maxHeight={context.maxHeight}
+        ref={ref}
+        tabIndex={-1}
+        testId={testId || 'popoverContent'}
+        theme={theme}
+        width={context.width}
+        data-popover-placement={context.positioning}
+        withoutPointer={context.withoutPointer}
+      >
+        <PopoverChildrenContent
+          aria-labelledby={context.popoverTriggerId.current}
+          role="dialog"
+          ref={focusTrapRef}
+        >
+          <StyledAnnounce maxHeight={context.maxHeight}>
+            {!content.length ? (
+              'Content must be passed'
```

Comment:
Not sure we should announce an "error" for the developers. I think we can assume there is content?